pandapipes/control/controller/leakage_controller.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2020 by Fraunhofer Institute for Energy Economics
+# and Energy System Technology (IEE), Kassel. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+import pandapipes as pp
+import numpy
+from pandapower.control.basic_controller import Controller
+
+try:
+    import pplog as logging
+except ImportError:
+    import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LeakageController(Controller):
+    """
+    Controller to consider a leak at pipes, valves or heat exchangers.
+
+    :param net: The net in which the controller resides
+    :type net: pandapipesNet
+    :param element: Element (first only "pipe", "valve", "heat_exchanger")
+    :type element: string
+    :param element_index: IDs of controlled elements
+    :type element_index: int[]
+    :param output_area_m2: Size of the leakage in m^2
+    :type output_area_m2: float[]
+    :param in_service: Indicates if the controller is currently in_service
+    :type in_service: bool, default True
+    :param recycle: Re-use of internal-data
+    :type recycle: bool, default True
+    :param drop_same_existing_ctrl: Indicates if already existing controllers of the same type and with the same matching parameters (e.g. at same element) should be dropped
+    :type drop_same_existing_ctrl: bool, default False
+    :param kwargs: Parameters for pipeflow
+    :type kwargs: dict
+
+    :Example:
+        >>> kwargs = {'stop_condition': 'tol', 'iter': 100, 'tol_p': 1e-7, 'tol_v': 1e-7, 'friction_model': 'colebrook',
+        >>>           'mode': 'hydraulics', 'only_update_hydraulic_matrix': False}
+        >>> LeakageController(net, element='pipe', element_index=0, output_area_m2=1, **kwargs)
+        >>> run_control(net)
+
+    """
+
+    def __init__(self, net, element, element_index, output_area_m2, profile_name=None,
```

Comment:
You should also be able to use something like kg_per_m_s meaning mass per meter and second.